samples/excel/82-document/get-file-in-slices-async.yaml

```diff
@@ -0,0 +1,201 @@
+id: excel-82-document-get-file-in-slices-async-
+name: Get file in slices and recombine
+description: Get current file in slices and recombine the slices afterwards.
+host: EXCEL
+api_set:
+    ExcelApi: 1.1
+script:
+    content: |-
+        $("#setup").click(() => tryCatch(setup));
+        $("#get-file").click(() => tryCatch(getCurrentFile));
+
+
+        function getCurrentFile() {
+            const sliceSize = 4096 /*Bytes*/
+
+            // Note: If you don't specify the slice size, a default will be assumed for you.  In this sample. to demonstrate how the slices work, a small size limit is specified.
+            Office.context.document.getFileAsync(Office.FileType.Compressed, { sliceSize: sliceSize },
+                function (result) {
+                    if (result.status === Office.AsyncResultStatus.Failed) {
+                        return onError(result.error);
+                    }
+
+                    // Result.value is the File object.
+                    getFileContents(result.value, onSuccess, onError);
+                });
+
+            function onError(error: Office.Error): void {
+                OfficeHelpers.UI.notify('Error', error.message);
+                OfficeHelpers.Utilities.log(error);
+            }
+
+            function onSuccess(byteArray: number[]) {
+                // Now that all the file content is stored in the "data" parameter,
+                // you can do something with it, such as print, store into database, etc.
+                console.log("Received the full contents of the file.");
+
+                let base64string = base64js.fromByteArray(byteArray);
+                $('#file-contents').val(base64string).show();
+
+                OfficeHelpers.UI.notify("You can validate the file contents in the text box using an online tool like https://www.base64decode.org/.  To do this, copy the base64 string and paste it into the tool decoder.  Decode and open the resulting file.");
```

Comment:
Suggest that you edit the text to read:

The base64-encoded string that represents the current document has been written to the text box below. To validate the string, use an online tool such as https://www.base64decode.org/ to decode the string, and then open the resulting file.

Reply:
Thanks, done.